.gitlab/ci/sdk-nightly.yml

```diff
@@ -1,6 +1,12 @@
 .sdk-nightly-schedule-rule:
   rules:
-    - if: '$CI_PIPELINE_SOURCE =~ /^(schedule)$/ && $DEMISTO_SDK_NIGHTLY == "true"'
+    - if: '$CI_PIPELINE_SOURCE =~ /^(schedule|trigger)$/ && $DEMISTO_SDK_NIGHTLY == "true"'
+
+# used for jobs which we want to run in a pipeline even when previous jobs in the pipeline fail e.g. slack notification
+.sdk-nightly-schedule-rule-always:
+  rules:
+    - if: '$CI_PIPELINE_SOURCE =~ /^(schedule|trigger)$/ && $DEMISTO_SDK_NIGHTLY == "true"'
+      when: always
```

Comment:
Can you extend the .sdk-nightly-schedule rule with always? might save a line of code duplication

Reply:
We don't want all `sdk-nightly`-related jobs to always run. Only the ones that slack-notification depend upon need to always execute regardless of the run-status of previous jobs in the pipeline.

Reply:
yeah I understand, I thought about something like this
```suggestion
# used for jobs which we want to run in a pipeline even when previous jobs in the pipeline fail e.g. slack notification
.sdk-nightly-schedule-rule-always:
  extends: .sdk-nightly-schedule-rule
  when: always 
```
but I'm not sure this would work with well other rules (applying the always to the specfic rule rather than all of them), maybe theres a syntax that does do the trick that you know of

Reply:
Ah - my bad, I misunderstood your comment. I don't think the `extends` syntax would work properly/as you expect as it's included in your suggestion. You could probably successfully use the `!reference` syntax but... ¯\_(ツ)_/¯